BootloaderCorePkg/Library/FspSupportLib/FspSupportLib.c

```diff
@@ -33,6 +33,54 @@ GetFspNvsDataBuffer (
   return GetGuidHobData (HobListPtr, Length, &gFspNonVolatileStorageHobGuid);
 }
 
+/**
+  Find an FSP resource descriptor large enough to accomodate
+  requested length.
+
+  @param  HobListPtr   A HOB list pointer.
+  @param  Length       A pointer to the requested length. If suitable
+                       descriptor is found, length will be updated with
+                       resource length.
+  @retval              Reserved region start address.  0 if this region does not exist.
+ */
+UINT64
+GetFspAvailableSystemMem(
+  CONST VOID     *HobListPtr,
+  UINT64         *Length
+)
+{
+  EFI_PEI_HOB_POINTERS    Hob;
+
+  /*
+   * Get the HOB list for processing
+   */
+  Hob.Raw = (VOID *)HobListPtr;
+
+  /*
+   * Collect memory ranges
+   */
+  while (!END_OF_HOB_LIST (Hob)) {
```

Comment:
Do we need to add any additional checks? Can this function find memory which is less than 1 MB?

Reply:
Guo and I are still discussing this fix. I think the three of us probably need to discuss this change some more.